src/ophyd_async/epics/pmac/_pmacTrajectory.py

```diff
@@ -0,0 +1,488 @@
+import time
+from typing import Any
+
+import numpy as np
+import numpy.typing as npt
+from pydantic import BaseModel, Field
+from scanspec.specs import Frames, Path, Spec
+from velocity_profile import velocityprofile as vp
+
+from ophyd_async.core import (
+    FlyerController,
+    wait_for_value,
+)
+from ophyd_async.epics.pmac import Pmac, PmacMotor
+
+TICK_S = 0.000001
+
+
+class PmacTrajInfo(BaseModel):
+    spec: Spec = Field(default=None)
```

Comment:
```suggestion
    spec: Spec[PmacMotor | Literal["DURATION"]] = Field(default=None)
```
So this makes me think we need to make duration special in scanspec

Reply:
This gives a pydantic error  ` /workspaces/ophyd-async/src/ophyd_async/epics/pmac/_pmacTrajectory.py:20:11 - error: "__getitem__" method not defined on type "type" (reportIndexIssue)`